### PR TITLE
Exclude articles with no images from articles lists #828

### LINKED ISSUE #828: FR: Exclude Articles Without Images from Magazine Feed Pages
### Context/Description

As a Mack Trucks magazine content author, I want articles without images to be automatically excluded from the magazine feed, so that the magazine feed only displays articles with visuals to provide the best user experience for my readers. 

### Acceptance Criteria

**User AC:**
- [ ] As a magazine reader, when I view the magazine feed each article displayed shows an image.

**Non-functional AC:**
- [ ] If an article is missing an image, in the [magazine-articles.xlsx](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/Shared%20Documents/sites/VolvoGroup/vg-macktrucks-com/[magazine-articles.xlsx](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/Shared%20Documents/sites/VolvoGroup/vg-macktrucks-com/magazine-articles.xlsx?d=wb2201e6b8c784eb7a2ebe2a58d352468&csf=1&web=1&e=e5JWDb)?d=wb2201e6b8c784eb7a2ebe2a58d352468&csf=1&web=1&e=e5JWDb) then it should be automatically removed from the magazine feed pages.
- [ ] Image URL is checked, if it is not present than the article does not appear in this block

### Out of scope

_No response_

### Note

Affected ticket - https://github.com/hlxsites/vg-macktrucks-com/issues/602

### Technical details

_No response_

### Attachment / link to Figma

_No response_

### PR DESCRIPTION
**Pull request by santi-homps**
Fix #828

This PR also fixes some minor issues an cleans a bit the code that affects the following blocks:

v2-explore-articles
v2-magazine-tabbed-carousel
explore-articles
recommendations
recent-articles
(these last 2 can be found on the old magazine template)

Some test pages with the affected blocks are:

/drafts/shomps/v2-explore-articles
/drafts/shomps/v2-magazine-tabbed-carousel
/magazine/explore-articles
/magazine/articles/2023/august/choosing-mack-is-a-no-brainer

Test URLs:

Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/
After: https://828-articles-no-images--vg-macktrucks-com--hlxsites.hlx.page/

### CODE REVIEW

File: scripts/magazine-helper.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
+import { getJsonFromUrl } from './common.js';
```
Comment (alexiscoelho) [inline scripts/magazine-helper.js:1]: Please rename this file to `magazine.service.js`
Put it inside a services folder, can be inside of scripts for now...

File: scripts/magazine-helper.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    return [];
+  }
+};
+
+/**
+ * Extracts the articles that dont have an image field
+ * @param {Array} articles - An array of articles
+ * @returns {Array} - The same array of articles but without those that dont have an image field
+ */
+export const removeArticlesWithNoImage = (articles) => {
```
Comment (alexiscoelho) [inline scripts/magazine-helper.js:34]: Take a close look at how you are calling this method on the files:
`removeArticlesWithNoImage(allArticles);`
And how your method looks.
You are returning the articles, but not using the return and changing the array by reference.
This in terms of programming is wrong and can lead to unexpected behaviours.
Instead, copy the array, do the changes, return the array and use the returned value. The provided array to this method should not change!

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: scripts/magazine-helper.js (hunk 1 above)
Comment (santi-homps) [reply scripts/magazine-helper.js:1]: Done

File: scripts/magazine-helper.js (hunk 2 above)
Comment (santi-homps) [reply scripts/magazine-helper.js]: Done!

**Review by alexiscoelho — APPROVED** [review]

**Review by amglebihan — APPROVED** [review]

**Pull request merged**